MyBlog.Services/Areas/Users/UserService.cs

```diff
@@ -1,53 +1,56 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
 using MyBlog.Infrastructure;
 using MyBlog.Infrastructure.Entities;
+using MyBlog.Service.Areas.Users.AutoMapper.Dto;
 using MyBlog.Service.Exception;
 
 namespace MyBlog.Service.Areas.Users;
 
 public class UserService : IUserService
 {
     private readonly MyBlogContext _context;
-
-    public UserService(MyBlogContext context)
+    private readonly IMapper _mapper;
+    public UserService(MyBlogContext context, IMapper mapper)
     {
         _context = context;
+        _mapper = mapper;
     }
 
-    public async Task<List<User>> GetListAsync()
+    public async Task<List<UserDtoGet>> GetListAsync()
     {
-        var users = await _context.Users.ToListAsync();
+        var users = await _context.Users.
+            Select(u => _mapper.Map<UserDtoGet>(u)).
+            ToListAsync();
         
         return users;
     }
 
-    public async Task<User> GetByIdAsync(int id)
+    public async Task<UserDtoGet> GetByIdAsync(int id)
     {
-        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id)
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id) 
                    ?? throw new NotFoundException($"User with Id: {id} is not found");
-
-        return user;
+        
+        var userDto = _mapper.Map<UserDtoGet>(user);
```

Comment:
you can replace this:

var user = await _context.Users
        .Select(u => _mapper.Map<UserDto>(u))
        .FirstOrDefaultAsync(u => u.Id == id)
        ?? throw new NotFoundException($"User with Id: {id} is not found");

Reply:
I disagree with this decision/